esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,349 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "ws_workunitsService.hpp"
+#include "ws_wudetails.hpp"
+
+void CWUDetails::process(IEspWUDetailsResponse &resp)
+{
+    unsigned __int64 maxTimestamp = 0;
+    IArrayOf<CAttributeFilter> arrayAttributeFilter;
+    UnsignedShortArray skReturnAttribList;
+    UnsignedShortArray nestedScopeTypes;
+    StringArray & ids = scopeFilter.getIds();
+    StringArray & scopes = scopeFilter.getScopes();
+    StringArray & scopeTypes = scopeFilter.getScopeTypes();
+
+    ids.pruneEmpty();
```

Comment:
It wasn't returning a const type so thought it would be ok to modify.  @afishbeck Please can you comment...is it okay to modify the input parameters returned by esp?